lessons/cryptography/cryptosystem-types/cryptosystems-types.md

```diff
@@ -0,0 +1,90 @@
+
+The strength of the Diffie-Hellman key exchange process depends on using a strong random number generator to choose the public modulus so that it cannot be easily predicted by an attacker. In addition, this process does not provide any method of authentication between Alice and Bob, so it is vulnerable to a man-in-the-middle attack where an attacker could intercept information between Alice and Bob and instead establish two separate Diffie-Hellman key exchanges with Alice and Bob. 
+
+### Elliptic Curve Cryptosystem
+Elliptic Curve Cryptosystems (ECC) are the most powerful type of cryptography currently used today. ECC is based on a mathematical concept called an elliptic curve, which is a set of points that satisfy a specific mathematical equation. For example, an elliptic curve is shown below. 
+
+![Sample Elliptic Curve (https://blog.cloudflare.com/a-relatively-easy-to-understand-primer-on-elliptic-curve-cryptography/)](.rsrc/elliptic-curve.png)
+
+ECC is based on solving a challenging mathematical problem that is considered more difficult to solve and more computationally expensive than the mathematical problem that RSA and Diffie-Hellman are based on. With ECC, smaller keys will achieve the same level of security compared to RSA. 
+
+## Symmetric vs Asymmetric Cryptography
+We previously discussed various symmetric and asymmetric cryptosystems. Below is a chart showing quick comparisons of some key features of both types of cryptosystems.
+
+| |Symmetric|Asymmetric|
+|----|----|----|
+|**Key**|Shared private key for encryption and decryption|Individual public key for encryption, individual private key for decryption|
+|**Speed**|Faster|Slower|
+|**Key Distribution**|Requires secure key distribution of the shared private key| Secure key distribution is not necessary since disclosure of public keys does not impact security of encrypted messages and private keys never need to be shared|
+|**Examples**|AES|RSA, Diffie-Hellman, ECC|
+
+
+# The Real World, Prolific Breaches
+In 2005, a cache-timing attack was performed on OpenSSL's AES encryption. It is known that side-channel attacks can occur, but this attack demonstrated obtaining the a complete AES private key. The time it takes to execute the AES algorithm on known plaintext messages can be used by attackers to to work backwards to determine the input. 
+
+# Check YoSelf
+## Q1 - Which cryptosystem is considered the most secure today? 
+a. Diffie-Hellman <br>
+b. AES <br> 
+c. RSA <br>
+d. ECC <---<br>
+
+## Q2 - T/F: Symmetric cryptosystems involve public and private keys for the sender and receiver. 
+a. True <br>
+b. False <---<br>
+
+## Q3 - Which of the following keys are used when Alice sends a message to Bob using asymmetric encryption? 
+a. Alice's public key and Alice's private key <br>
+b. Bob's public key and Bob's private key <---<br>
```

Comment:
good but lets phrase these as they key and what its used for, so something like:
b. Alice uses Bob's public key to encrypt the message, then Bob uses his private key to decrypt the message

Reply:
Updated